docs: point users to docs.cilium.io in GH docs

aanm opened the pull request:
Since users might read the github docs we will provide them a warning
in the documentation, that will not show up in the generated html code
but it shows up in github, to point them into the stable documentation
in http://docs.cilium.io

Signed-off-by: André Martins <andre@cilium.io>

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/4627)
<!-- Reviewable:end -->

raybejjani reviewed (approved):
Do we expect deeplinks to the GH pages? most would land on the main README, wouldn't they?

I'm not opposed to this but I'm surprised anyone was expecting a GH page to be stable... shrug

aanm committed:
docs: point users to docs.cilium.io in GH docs

Since users might read the github docs we will provide them a warning
in the documentation, that will not show up in the generated html code
but it shows up in github, to point them into the stable documentation
in http://docs.cilium.io

Signed-off-by: André Martins <andre@cilium.io>
